VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+        $isLegacyWldnsxT = Get-VCFNsxtCluster | Where-Object {$_.domains.id -eq $domain.id}
+        $isLegacyWldnsxV = Get-VCFNsxvManager | Where-Object {$_.domain.id -eq $domain.id}
+    }
+    elseIf (([float]$sddcMgrVersion -ge [float]4) -AND ($domainType -eq "Management"))
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.type -eq "MANAGEMENT" }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    else
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    Foreach ($vCenter in $vCenterServer) {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $vCenter.fqdn
+            'name'       = $vCenter.fqdn.split(".")[0]
+            'resourceId' = $vCenter.id
+            'type'       = "VCENTER"
+        }
+    }
+
+    # NSX V Manager
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management") -OR ($isLegacyWldnsxV))
+    {
+        $nsxvManager = Get-VCFNsxvManager | Where-Object { $_.domain.id -eq $domain.id }
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $nsxvManager.fqdn
+            'name'       = $nsxvManager.fqdn.split(".")[0]
+            'resourceId' = $nsxvManager.id
+            'type'       = "NSX_MANAGER"
+        }
+    }
+
+    #VRLI
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $logInsight = Get-VCFvRLI *>$null
+        If ($logInsight)
+        {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = ($logInsight.nodes | Where-Object {$_.type -eq "MASTER"}).fqdn
+                'name'       = (($logInsight.nodes | Where-Object {$_.type -eq "MASTER"}).fqdn).split(".")[0]
+                'resourceId' = $logInsight.id
+                'type'       = "VRLI"
+            }    
+        }
+    }
+    
+    #PSCs
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $pscs = Get-VCFPsc | Where-Object { $_.domain.id -eq $domain.id }
+        Foreach ($psc in $pscs) {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = $psc.fqdn
+                'name'       = $psc.fqdn.split(".")[0]
+                'resourceId' = $psc.id
+                'type'       = "PSC"
+            }
+        }  
+    }
+
+    #NSXT
+    If (([float]$sddcMgrVersion -ge [float]4) -OR ($isLegacyWldnsxT))
+    {
+        $nsxtManager = Get-VCFNsxtCluster | Where-Object { $_.domains.id -eq $domain.id }
+        $nsxtSans = @()
+        Foreach ($nodeFqdn in $nsxtManager.nodes.fqdn) 
+        {
```

Comment:
```suggestion
        Foreach ($nodeFqdn in $nsxtManager.nodes.fqdn)  {
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)